### PR TITLE
Add metrics support

### PR DESCRIPTION
**Pull request by bswartz**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Add metrics support to the library

**Does this PR introduce a user-facing change?**:
```release-note
The library now collects metrics internally and can expose them through an HTTP listener if the importing application desires. The metrics include a histogram of populator operation durations (and their results) as well as a gauge of the number of populator operations in flight.
```

### CODE REVIEW

**Comment by xing-yang** [comment]
In the release note, can you clarify what metrics are collected?

**Comment by xing-yang** [comment]
There are CI errors. PTAL.

File: populator-machinery/metrics.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+	err := m.srv.Shutdown(context.Background())
+	if err != nil {
+		klog.Errorf("Failed to shutdown metrics server: %s", err.Error())
+	}
+
+	klog.Infof("Metrics server successfully shutdown")
+}
+
+// OperationStart starts a new operation
```
Comment (xing-yang) [inline populator-machinery/metrics.go:137]: nit: operationStart

File: populator-machinery/metrics.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	if _, exists := m.cache[pvcUID]; !exists {
+		m.cache[pvcUID] = time.Now()
+	}
+	m.opInFlight.Set(float64(len(m.cache)))
+}
+
+// OperationStart drops an operation
```
Comment (xing-yang) [inline populator-machinery/metrics.go:148]: dropOperation

File: populator-machinery/metrics.go [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+func (m *metricsManager) dropOperation(pvcUID types.UID) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	delete(m.cache, pvcUID)
+	m.opInFlight.Set(float64(len(m.cache)))
+}
+
+// RecordMetrics emits operation metrics
```
Comment (xing-yang) [inline populator-machinery/metrics.go:156]: nit: recordMetrics

File: populator-machinery/metrics.go (hunk 1 above)
Comment (bswartz) [reply populator-machinery/metrics.go]: Fixed

File: populator-machinery/metrics.go (hunk 2 above)
Comment (bswartz) [reply populator-machinery/metrics.go]: FWIW, this is a copy paste from external-snapshotter, which has the same typo

File: populator-machinery/metrics.go (hunk 3 above)
Comment (bswartz) [reply populator-machinery/metrics.go]: Fixed

**Comment by bswartz** [comment]
> In the release note, can you clarify what metrics are collected?

Done

File: populator-machinery/metrics.go (hunk 2 above)
Comment (xing-yang) [reply populator-machinery/metrics.go]: ok:)

**Pull request merged**